### ISSUE TITLE
QA: Accessibility

### ISSUE THREAD

**Issue by alanmoo** [mozilla/donate-wagtail#257]
Master QA ticket for site accessibility

If an item is not completely passing, file a new ticket and edit this post to link to it. For example:
- [x] A thing that should work, but doesn't (#1234)

Any QA tickets that don't show a reference from a master issue will be closed.

- [x] Run accessibility test (tbd; @youriwims can you make a suggestion?) on all pages and ensure there are no glaring omissions
  - [ ] Main donate page`
    - [x] ~Enable Zooming + Scaling (#302) 🆘~
    - [x] Add aria role's for nav items (#304) 🆘 
    - [ ] Improve donate footer links color contrast (#306) ⚠️ 
    - [ ] Update aria roles for radio button group (#315) 🔍   
    - [x] Revise semantic structure for header elements (#316) ⚠️ 
    - [ ] Update radio button[name="donation_frequency"] group semantic structure/aria labels (#317) ⚠️ 
    -  [ ] Update radio button[name="amount"] group semantic structure/aria labels (#318) ⚠️ 
    - [x] Refine main page donate error state (#336) ⚠️ 
    - [ ]  Create clear error state of 'Other Amount' <input> (#363) 🔍  
    - [ ] Refine menu/burger focus state (#340) 🔍 
  - [ ] A campaign page
    - _Any issues it has are already covered in some of these other tickets._
  - [ ] Credit card form
    - [x] Improve contrast ratio of 'Change Amount' link (#319) ⚠️ 
    -  [x] ~Fix autocomplete attribute values (#320) ⚠️~
    - [ ] Email input on cc payment form presents no error for invalid input (#366) ⚠️ 
    - [ ] Zip code has no error for invalid input (#367) ⚠️  
  - [ ] Upsell view
    -  [ ] Fix contrast ratio for upsell form <input> (#328) ⚠️  
  - [ ] Newsletter signup
    - [ ] Privacy Notice link fails required AA contrast ratio (#313) ⚠️ 
    - [ ] Improve footer form's focus state (#312) ⚠️ 
  - [ ] Thank you page/Social share
    -  [ ] Thank You form button needs more contrast (#322) 🔍
    - [ ] Add <title> text for Social share (#327) ⚠️ 

#### Symbol Key
🆘 - Critical [P1]
⚠️ - Moderate/Serious [P2]
🔎 - '_Take a look into this when you have a chance..._' [P3]

**Comment by youriwims** [comment]
Most definitely—I recommend that we use a combination of automated + manual testing 🙏. [Axe-cli](https://github.com/dequelabs/axe-cli) comes highly recommended by a11y professionals/specialists for automated testing.

I've had really good results with the Accessibility Insights for Web (AIW) browser extension for local dev. Depending on the amount of time that we'd like to spend on the accessibility assessment, there's an 'in-depth' assessment and a 'fast pass' test. Although if we implement axe-cli, I think it'd be okay if we only focused on certain sections of the in-depth assessment for manual testing. _See examples of the different testing options with AIW below_

- - -
### **Fast Pass Assessment**
![Screen Shot 2019-09-23 at 2 01 52 PM](https://user-images.githubusercontent.com/36484608/65450398-00be7880-de0b-11e9-9017-6f2122872ffc.png)
- - -
### **In-depth Guided Assessment**
![Screen Shot 2019-09-23 at 2 02 10 PM](https://user-images.githubusercontent.com/36484608/65450420-10d65800-de0b-11e9-8069-fd314b1c4abf.png)
- - -

**Comment by youriwims** [comment]
@alanmoo can you give me more deets for when you say 'A campaign page' in the initial comment? Do you want tests done on just any random campaign page?...or are there campaign pages specific to donate qa atm?

**Comment by alanmoo** [comment]
https://donate-wagtail.mofostaging.net/en-US/fighting-back/

**Comment by jamilasnell** [comment]
We need to re-access accessibility
